# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/ContentParent.h
# Commit: 43cd1d6eb145
# Full Hash: 43cd1d6eb1459ae16207f4c9396cf5113a7b034a
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2022-01-11 03:49:34
# Regressor Bug: 1737828
# File Overlap Count: 2
# Description:
#   Bug 1737828 - Avoid the main thread when starting PBackground, r=asuth,ipc-reviewers,necko-reviewers,kershaw,handyman
#   
#   This patch avoids requiring the main thread to create PBackground instances by
#   instead using a background starter TaskQueue, and sending messages from a
#   PBackgroundStarter actor hosted on that thread to the target background thread
# ==============================================================================

diff -r 400d19ddbff7 -r 43cd1d6eb145 dom/ipc/ContentParent.h
--- a/dom/ipc/ContentParent.h	Mon Jan 10 19:44:23 2022 +0000
+++ b/dom/ipc/ContentParent.h	Mon Jan 10 20:09:14 2022 +0000
@@ -917,7 +917,7 @@
       const nsACString& aContentProcessType);
 
   mozilla::ipc::IPCResult RecvInitBackground(
-      Endpoint<mozilla::ipc::PBackgroundParent>&& aEndpoint);
+      Endpoint<mozilla::ipc::PBackgroundStarterParent>&& aEndpoint);
 
   mozilla::ipc::IPCResult RecvAddMemoryReport(const MemoryReport& aReport);
   mozilla::ipc::IPCResult RecvAddPerformanceMetrics(